v1.0 function documentation

benmicrosoft opened the pull request:
This PR includes documentation updates for our GA to v1.0 for the VA get function

benmicrosoft commented:
https://microsoftgraph.visualstudio.com/Roadmap/_workitems/edit/16973

JarbasHorst commented on api-reference/v1.0/api/virtualappointment-getvirtualappointmentjoinweburl.md:
```diff
@@ -0,0 +1,85 @@
+
+| Permission type                        | Permissions (from least to most privileged)                   |
+|:---------------------------------------|:--------------------------------------------------------------|
+| Delegated (work or school account)     | VirtualAppointment.Read, VirtualAppointment.ReadWrite         |
```
The delegated permissions listed here are different than the delegated permissions listed in the corresponding BETA topic => api-reference/beta/api/virtualappointment-getvirtualappointmentjoinweburl.md

Is this correct?

benmicrosoft replied:
Correct, graph council recommended not brining the onlinemeeting scopes to v1.0

Jarbas-MSFT commented:
Hi @benmicrosoft, I reviewed your PR and left one comment, please take a look when you will have time.

I can't find you in Teams. Can you please share with me there the CSDL and api.md files for this PR?

Jarbas-MSFT commented on api-reference/beta/api/virtualappointment-getvirtualappointmentjoinweburl.md:
```diff
@@ -13,16 +13,17 @@ Namespace: microsoft.graph
 [!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
 
-Get a join web URL for [Microsoft Virtual Appointments](/microsoft-365/frontline/virtual-appointments). This web URL includes enhanced business-to-customer experiences such as mobile broswer join and virtual lobby rooms.
+Get a join web URL for [Microsoft Virtual Appointments](/microsoft-365/frontline/virtual-appointments). This web URL includes enhanced business-to-customer experiences such as mobile browser join and virtual lobby rooms. With [Teams Premium](/microsoftteams/teams-add-on-licensing/licensing-enhance-teams), you can configure a [custom lobby room](/microsoft-365/frontline/browser-join#customize-the- ...
```
Was there a specific reason for ` view=o365-worldwide` in the custom lobby room URL? I removed it because it was causing the API Doctor validation to fail.

benmicrosoft replied:
No, we can remove. Thanks!

JarbasHorst commented on api-reference/v1.0/api/virtualappointment-getvirtualappointmentjoinweburl.md:
```diff
@@ -0,0 +1,85 @@
+<!-- {
+  "blockType": "response",
+  "truncated": true,
+  "@odata.type": "microsoft.graph.virtualAppointment"
```
Since this API returns a string, I think we should remove the odata.type property from this HTML comment; otherwise, API Doctor will validate the response object against a virtualAppointment object that doesn't exist in v1.0.

Same here:
- api-reference/beta/api/virtualappointment-getvirtualappointmentjoinweburl.md

benmicrosoft replied:
updated

JarbasHorst commented:
Hi @Lauragra, this PR is ready for final review.

Can you or @benmicrosoft help resolve the merge conflicts? I don't have enough permissions on the repository to do that.

benmicrosoft commented:
> Hi @benmicrosoft, yes, I refactored the documentation for the getVirtualAppointmentJoinWebUrl method. For consistency, it should be listed under Online Meeting, since it is a method of the onlineMeeting resource. For example, the same also applies to the createOrGet method. Is there a specific reason why you want to highlight getVirtualAppointmentJoinWebUrl?
> 
> ![image](https://user-images.githubusercontent.com/26866873/236501939-a9b12097-051a-45f6-af41-d6fefce621e3.png) ![image](https://user-images.githubusercontent.com/26866873/236502076-af6efdcc-309d-41f9-86aa-978355bab3cf.png)

Got it. For a bit of context, Virtual appointment is currently live on Beta with its own resource and method that is being deprecated and replaced with this get function. In the fall, we are going to reintroduce Virtual appointment with new storage and bring back the CRUD methods and resource. @Lauragra and I had decided we should keep Virtual appointment with it's own node, since even though it is currently only a Get function of Online Meeting, we will move this eventually to be under Virtual appointment when we expand the functionality. 

My preference is that we keep it separated out but will defer to you two @JarbasHorst @lauragra on the right approach.

JarbasHorst commented:
Hi @benmicrosoft, thanks for the clarification. 

If for historical reasons this method should be under Virtual appointment, then let's move it back. However, I don't think that this is the reason for the merge conflicts. Can you try to resolve the conflicts, please?

benmicrosoft commented:
hi @JarbasHorst the error should now be resolved

srikhs commented on api-reference/beta/resources/onlinemeeting.md:
```diff
@@ -15,18 +15,19 @@ Namespace: microsoft.graph
 | [Delete](../api/onlinemeeting-delete.md) | None | Delete an **onlineMeeting** object. |
 | [Create or get onlineMeeting](../api/onlinemeeting-createorget.md) | [onlineMeeting](onlinemeeting.md) | Create an online meeting with a custom, external ID. If the meeting already exists, retrieve its properties. |
 | [List transcripts of an onlineMeeting](../api/onlinemeeting-list-transcripts.md) | [callTranscript](callTranscript.md) collection | Retrieve the list of transcripts of an **onlineMeeting**. |
+| [Get virtual appointment join link](../api/onlinemeeting-getvirtualappointmentjoinweburl.md) | String | Get a join web URL for a [Microsoft Virtual Appointments](/microsoft-365/frontline/virtual-appointments). |
```
we don't need this anymore, since we are adding it only under Virtual Appointment.

srikhs commented on api-reference/v1.0/resources/onlinemeeting.md:
```diff
@@ -22,6 +22,7 @@ Contains information about a meeting, including the URL used to join a meeting,
 | [Update](../api/onlinemeeting-update.md)                           | [onlineMeeting](onlinemeeting.md) | Update the properties of an **onlineMeeting** object. |
 | [Delete onlineMeeting](../api/onlinemeeting-delete.md)             | None                              | Delete an **onlineMeeting** object.                                                                                    |
 | [Create or get onlineMeeting](../api/onlinemeeting-createorget.md) | [onlineMeeting](onlinemeeting.md) | Create an **onlineMeeting** object with a custom, external ID. If the meeting already exists, retrieve its properties. |
+| [Get virtual appointment join link](../api/onlinemeeting-getvirtualappointmentjoinweburl.md) | String | Get a join web URL for a [Microsoft Virtual Appointments](/microsoft-365/frontline/virtual-appointments). |
```
we don't need this anymore since we are adding under Virtual Appointment.

srikhs replied:
virtualappointment

srikhs commented on api-reference/beta/resources/onlinemeeting.md:
```diff
@@ -15,18 +15,19 @@ Namespace: microsoft.graph
 | [Delete](../api/onlinemeeting-delete.md) | None | Delete an **onlineMeeting** object. |
 | [Create or get onlineMeeting](../api/onlinemeeting-createorget.md) | [onlineMeeting](onlinemeeting.md) | Create an online meeting with a custom, external ID. If the meeting already exists, retrieve its properties. |
 | [List transcripts of an onlineMeeting](../api/onlinemeeting-list-transcripts.md) | [callTranscript](callTranscript.md) collection | Retrieve the list of transcripts of an **onlineMeeting**. |
+| [Get virtual appointment join link](../api/onlinemeeting-getvirtualappointmentjoinweburl.md) | String | Get a join web URL for a [Microsoft Virtual Appointments](/microsoft-365/frontline/virtual-appointments). |
```
virtualappointment?

benmicrosoft replied:
I removed this from v1.0 and beta

Jarbas-MSFT commented:
Hi @Lauragra, this PR is ready for final review.
